Eastern-Research-Group/expert-query
# Feature/350 switch to vite

## Related Issues:
* [EQ-350](https://jira.epa.gov/browse/EQ-350)
* [EQ-441](https://jira.epa.gov/browse/EQ-441)

## Main Changes:
* Switched from create-react-app to vite.
* Fixed issue of 404.html and 500.html pages not loading `epa-template-files` resources.
* Updated the db ssl logic to work of an environment variable instead of NODE_ENV to make it easier to connect to dev/stage/prod databases from local env.
* Added server side code coverage.
  * Note: I deleted the `api-request.cy.ts` tests, because I moved these to the server side tests. When these tests were in cypress they didn't really impact coverage much.
* Added more tests to get code coverage above 80%.

## Steps To Test:
1. Add `DB_SSL=false` to the `etl/.env.local` file
2. Run the etl locally and verify it works
3. Run `npm run setup` from app folder
4. Add `DB_SSL=true` to the `app/server/.env.local` file
5. Update other variables to point your local environment to the dev database
6. Open an ssh tunnel to the dev database
7. Run `npm run dev` to start the app
8. Manually test various features
9. Run `npm run coverage` 
10. Verify all tests pass
11. Verify coverage results are above 80% and include server side code
